exporting/prometheus/prometheus.c

```diff
@@ -326,6 +326,47 @@ void format_host_labels_prometheus(struct instance *instance, RRDHOST *host)
     rrdlabels_walkthrough_read(host->rrdlabels, format_prometheus_label_callback, &tmp);
 }
 
+/**
+ * Check the input buffer for the label name and maybe remove the label.
+ *
+ * @param b the buffer to check.
+ * @param name the label name to search for.
+ * @param keepExistingLabel zero to delete the existing label, non-zero otherwise.
+ * @return Returns 1 if the label exists and is kept. Returns 0 in all other cases.
+ */
+static int check_existing_label(BUFFER *b, const char *name, int keepExistingLabel) {
+    char k[PROMETHEUS_ELEMENT_MAX + 2];
+    size_t len = prometheus_name_copy(k, name, PROMETHEUS_ELEMENT_MAX);
+    k[len++] = '=';
+    k[len] = '\0';
+    const char *end = b->buffer + b->len;
+    char *match = strstr(b->buffer, k);
+    if (match == NULL)
+        return 0;
+    else if (keepExistingLabel)
+        return 1;
+    const char *matchEnd = strchr(match, ',');
+    if (matchEnd != NULL) {
+        // Have subsequent label(s), shift them to discard the matched label.
+        ++matchEnd;
+        memmove(match, matchEnd, end - matchEnd);
```

Comment:
I have a concern here that it is possible we do not have enough buffer to copy.  :thinking: 
As you can see in [buffer_fast_strcat](https://github.com/netdata/netdata/blob/56499814dbbe6b6a82235f32e1c4ca2d142b0c28/libnetdata/buffer/buffer.h#L203-L228) we call inside this function another function named [buffer_need_bytes](https://github.com/netdata/netdata/blob/56499814dbbe6b6a82235f32e1c4ca2d142b0c28/libnetdata/buffer/buffer.h#L151-L154) before to copy data. I suggest you also call it before to copy the data.

Reply:
This condition means that we have labels _after_ the label we wish to discard, so I move the later labels back.

As an example, if we have a sequence:
```
chart="abc",dimension="def",family="ghi",foo="bar"
``` 
When we attempt to remove family, this memmove on line 352 is moving the `foo="bar"` back to replace family. In other words, I am shortening the string length, not extending it. The write is still using your original `buffer_sprintf`

```
chart="abc",dimension="def",foo="bar"
```